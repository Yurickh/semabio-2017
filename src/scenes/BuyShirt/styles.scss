@import 'ambience/color';
@import 'ambience/responsive';
@import 'base/mask-image';
@import 'base/responsive-prop';

.page-shirt {
  width: 66vw;
  margin: 10vh auto;

  color: $color-emperor;

  & > .arrow {
    width: 17px;
    height: 14px;

    cursor: pointer;

    background-color: $color-emperor;

    @include mask-image('./arrow-back.svg');
  }

  & > .title {
    margin-top: 1em;

    @include responsive-prop(font-size, 30, 30);
  }

  & > .title > span {
    text-transform: uppercase;
  }

  & > .variant-selector {
    user-select: none;

    @include responsive-prop(margin-top, 26, 26);
  }

  & > footer {
    display: flex;
    align-items: center;
    justify-content: space-between;

    @include small-only {
      flex-direction: column;

      height: 10em;
    }
  }

  & > footer > .simple-button {
    @include responsive-prop(width, 182, 182);
    @include responsive-prop(height, 55, 55);
    @include responsive-prop(font-size, 18, 18);
  }

  & > footer > .simple-button.-disabled {
    cursor: not-allowed;
    pointer-events: none;

    opacity: .4;
    background-color: $color-emperor;

    &:hover {
      background-color: transparent;
    }
  }
}
