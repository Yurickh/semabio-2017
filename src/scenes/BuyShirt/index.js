import React, { Component } from 'react'

import ShopClient from '../../helpers/shopClient'
import GraphQL from '../../helpers/GraphQL'

import Button from '../../components/Button'
import VariantSelector from '../../components/VariantSelector'

import './styles.css'

class BuyShirt extends Component {
	state = {
		selectedShirtSize: null,
		ready: false,
	}

	redirectToHome = () => {
		window.location.href = '/'
	}

	componentWillMount() {
		const { product } = this.props.match.params
		if (product && !['C', 'D', 'E'].includes(product)) {
			return this.redirectToHome()
		}
	}

	goBack = () => {
		const { product } = this.props.match.params
		window.location.href = `/comprar/${product}`
	}

	refreshShirtSize = ([selectedShirtSize]) => {
		this.setState({ selectedShirtSize })
	}

	loading = () => {
		this.setState({
			ready: false
		})

		setTimeout(() => {
			this.setState({
				ready: true
			})
		}, 2000)
	}

	finishHim = () => {
		const { checkoutId } = this.props.match.params
		const shopClient = ShopClient()

		this.loading()

		const lineItems = {
			variantId: this.state.selectedShirtSize.id,
			quantity: 1,
		}

		if (checkoutId) {
			shopClient.addLineItems(checkoutId, lineItems)
			.then(checkout => {
				window.location.href = GraphQL.read(checkout).get('webUrl')
			})
		} else {
			shopClient.createCheckout()
			.then(checkout => {
				const checkoutId = GraphQL.read(checkout).get('id')

				return shopClient.addLineItems(checkoutId, lineItems)
			})
			.then(checkout => {
				window.location.href = GraphQL.read(checkout).get('webUrl')
			})
		}
	}

	onLoad = () => {
		this.setState({
			ready: true
		})
	}

	render() {
		const { product } = this.props.match.params
		return (
			<div className="page-shirt">
				<div className="arrow" onClick={this.goBack} />

				<h1 className="title">
					{
						this.props.match.params.checkoutId
						? 'Agora só falta você escolher o tamanho da sua camiseta :)'
						: 'Escolha o tamanho da sua camiseta :)'
					}
				</h1>

				<VariantSelector
					product="shirt"
					package={product}
					currentColor="#58C166" // color-forest
					onChange={this.refreshShirtSize}
					onLoad={this.onLoad}
					allowedToSelect={true}
				/>

				<footer>
					<span>ATENÇÃO: Não haverá reembolso em caso de desistência</span>
					<Button
						color='green'
						onClick={this.finishHim}
						onLoad={this.onLoad}
						loading={!this.state.ready}
						className={!this.state.selectedShirtSize ? '-disabled' : ''}>
						Finalizar Compra
					</Button>
				</footer>
			</div>
		)
	}
}

export default BuyShirt
