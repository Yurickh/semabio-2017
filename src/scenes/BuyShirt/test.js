import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import BuyShirt from '../BuyShirt'

describe('BuyShirt', () => {
  it('should render correctly', () => {
    const tree = shallow(<BuyShirt
      match={{
        params: {
          product: 'shirt'
        }
      }}
    />)
    expect(toJSON(tree)).toMatchSnapshot()
  })
})
