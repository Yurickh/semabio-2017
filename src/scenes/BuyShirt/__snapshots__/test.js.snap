// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyShirt should render correctly 1`] = `
<div
  className="page-shirt"
>
  <div
    className="arrow"
    onClick={[Function]}
  />
  <h1
    className="title"
  >
    Escolha o tamanho da sua camiseta :)
  </h1>
  <VariantSelector
    currentColor="#58C166"
    onChange={[Function]}
    package="shirt"
    product="shirt"
  />
  <footer>
    <span>
      ATENÇÃO: Não haverá reembolso em caso de desistência
    </span>
    <Button
      className="-disabled"
      color="green"
      onClick={[Function]}
    >
      Finalizar Compra
    </Button>
  </footer>
</div>
`;
